iothub/device/src/Transport/AmqpIoT/AmqpIoTConnector.cs

```diff
@@ -49,14 +49,9 @@ public async Task<AmqpIoTConnection> OpenConnectionAsync(TimeSpan timeout)
                 MaxFrameSize = AmqpConstants.DefaultMaxFrameSize,
                 ContainerId = CommonResources.GetNewStringGuid(),
                 HostName = _hostName,
+                IdleTimeOut = Convert.ToUInt32(_amqpTransportSettings.IdleTimeout.TotalMilliseconds),
             };
 
-            TimeSpan idleTimeout = _amqpTransportSettings.IdleTimeout;
-            if (idleTimeout != null)
```

Comment:
TimeSpan cannot be null

Reply:
There is `TimeSpan.Zero` though.

Reply:
True, but that would mean that the user deliberately set this value; which even though is weird, we'd allow.